pkg/webhooks/machine_webhook.go

```diff
@@ -922,6 +923,13 @@ func defaultGCP(m *machinev1.Machine, config *admissionConfig) (bool, []string,
 
 	providerSpec.Disks = defaultGCPDisks(providerSpec.Disks, config.clusterID)
 
+	if len(providerSpec.GPUs) != 0 {
+		// In case Count was not set it should default to 1, since there is no valid reason for it to be purposely set to 0.
```

Comment:
Will we have a gpu attached to each newly created wm?

Reply:
This will only kick in when a GPU instance is specified by a customer, so no, not every instance